ingest-app/src/cmr/ingest/services/granule_bulk_update/s3/umm_g.clj

```diff
@@ -26,8 +26,20 @@
         s3-urls (urls->s3-urls urls)]
     (concat s3-urls other-urls)))
 
+(defn- appended-related-urls
+  "Take the RelatedUrls, add any unique s3 urls, updating already existing urls"
+  [related-urls urls]
+  (let [s3-urls (urls->s3-urls urls)]
+    (concat related-urls s3-urls)))
```

Comment:
This will cause existing s3-url be duplicated if it is provided in the append list again.